docs/DeveloperGuide.md

```diff
@@ -1,33 +1,212 @@
+architecture. The below figure shows the command classes available and they are called based on the 
+`BookmarkParser` class as illustrated in Figure 2. 
+
+![BookmarkCommand ClassDiagram](Images/bookmarkCommand_Class.png)
+
+Figure 4 shows a more detailed sequence diagram of how the `BookmarkCategory`, `BookmarkCommand` 
+and `BookmarkList` interacts with each other for the scenario when the user input an `AddLinkCommand` 
+in Bookmark mode.
+
+![BookmarkCommand ClassDiagram](Images/AddCommand_sequenceDiagram.png)
+
+### **Academic Component**
+
+This section will describe in detail how some features inside the academic tracker section have been implemented.
+Figure x illustrates the general overview, the associations and the multiplicity of the academic classes.
+
+![Academic_Class_Diagram](Images/Academic_Class_Diagram.png)
+
+API: `java.academic`
+
+The above diagram looks at the overall structure of how the academic tracker is being implemented. 
+This component is split into 7 different classes, 
+their associations and multiplicity as explained in the above diagram x. 
+The functions of the academic tracker will be called through the `AcademicRun` class 
+when the program is in academic mode, which will subsequently call 
+the functions in `PersonBook` or `GradeBook`. 
+
+The academic component:
+* initialises two arraylists, `ArrayList<Grade>` and `ArrayList<People>` 
+to store the relevant `Grade` and `People` objects.
+* uses `AcademicCommandParser` to parse the user command.
+* identifies `AcademicCommandType` to decide 
+which of the commands under `PersonBook` or `GradeBook` is to be executed.
+* calls `AcademicStorage` to store the current set of data into the local storage file.
+
+**Academic Implementation**
+
+This section explains the details on how certain features 
+are implemented in the academic tracker.
+
+**GradeBook Features**
+The grade features are facilitated `Gradebook`, which further make use of `Grade`. 
+All grades are stored internally under `AcademicRun` as an array list `ArrayList<Grade>`.
+It implements the following operations:
+* `addGrade(String[], ArrayList<Grade>)`:Adds a `Grade` to the `ArrayList<Grade>`. 
+* `printCap(ArrayList<Grade>)`:Calculate the current CAP based on the `ArrayList<Grade>`. 
+* `printListOfGrades(ArrayList<Grade>)`:Print out all the `Grade` that are currently stored inside `ArrayList<Grade>`.
+* `deleteGrade(Integer, ArrayList<Grade>)`:Delete a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `suGradeInGradeBook(Integer, ArrayList<Grade>)`:Su a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `starGrade(Integer, ArrayList<Grade>)`:Star a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `combineGradeDetails(Grade)`:Print out the details of a particular `Grade`.
+
+### **Flashcard Component**
+
+This section will describe in detail how the flashcard feature is implemented.
+
+![Flashcard Component Class Diagram](Images/Flashcard_Class.png)
+
```

Comment:
Some arrows are not quite clear. Maybe you can point out both directions of multiplicity.

Reply:
Are Flashcard and FlashcardSotorage abstract? Would be nice if you specify using the label {abstract}, if they are not, you can remove the _italics_.